DSCResources/MSFT_xDefaultGatewayAddress/MSFT_xDefaultGatewayAddress.psm1

```diff
@@ -89,175 +212,171 @@ function Test-TargetResource
-            }
+            Write-Verbose -Message ( @("$($MyInvocation.MyCommand): "
+                $($LocalizedData.DefaultGatewayExistsButShouldNotMessage)
+                ) -join '' )
+            $desiredConfigurationMatch = $false
         }
         else
         {
-            Write-Verbose -Message "Default Gateway address is in the desired state."
-            return $true
+            Write-Verbose -Message ( @("$($MyInvocation.MyCommand): "
+                $($LocalizedData.DefaultGatewayExistsAndShouldMessage)
+                'Default Gateway does not exist which is correct.'
+                ) -join '' )
         }
-
     }
-    catch
+
+    return $desiredConfigurationMatch
+}
+
+#######################################################################################
+#  Helper functions
+#######################################################################################
+function Test-ResourceProperty {
+    # Function will check the Address details are valid and do not conflict with
+    # Address family. Ensures interface exists.
+    # If any problems are detected an exception will be thrown.
+    [CmdletBinding()]
+    param
+    (
+        [String]$Address,
+
+        [Parameter(Mandatory)]
+        [ValidateNotNullOrEmpty()]
+        [String]$InterfaceAlias,
+
+        [ValidateSet('IPv4', 'IPv6')]
+        [String]$AddressFamily = 'IPv4'
+    )
+
+    if (-not (Get-NetAdapter | Where-Object -Property Name -EQ $InterfaceAlias ))
     {
-        Write-Verbose -Message ( @(
-            "Can not set or find valid Default Gateway address using InterfaceAlias $InterfaceAlias and "
-            "AddressFamily $AddressFamily"
-            ) -join ""
-        )
-        throw $_.Exception
+        $errorId = 'InterfaceNotAvailable'
+        $errorCategory = [System.Management.Automation.ErrorCategory]::DeviceError
+        $errorMessage = $($LocalizedData.InterfaceNotAvailableError) -f $InterfaceAlias
+        $exception = New-Object -TypeName System.InvalidOperationException `
+            -ArgumentList $errorMessage
+        $errorRecord = New-Object -TypeName System.Management.Automation.ErrorRecord `
+            -ArgumentList $exception, $errorId, $errorCategory, $null
+
+        $PSCmdlet.ThrowTerminatingError($errorRecord)
     }
-}
+    if ($Address)
+    {
+        if (-not ([System.Net.Ipaddress]::TryParse($Address, [ref]0)))
+        {
+            $errorId = 'AddressFormatError'
+            $errorCategory = [System.Management.Automation.ErrorCategory]::InvalidArgument
+            $errorMessage = $($LocalizedData.AddressFormatError) -f $Address
+            $exception = New-Object -TypeName System.InvalidOperationException `
+                -ArgumentList $errorMessage
+            $errorRecord = New-Object -TypeName System.Management.Automation.ErrorRecord `
+                -ArgumentList $exception, $errorId, $errorCategory, $null
+
+            $PSCmdlet.ThrowTerminatingError($errorRecord)
+        }
```

Comment:
nit: could we add a line break here and after $detectedAddressFamily it'd make it a bit easier to read.